plugin/pkg/scheduler/algorithm/priorities/taint_toleration.go

```diff
@@ -57,6 +57,12 @@ func ComputeTaintTolerationPriorityMap(pod *v1.Pod, meta interface{}, nodeInfo *
 	if node == nil {
 		return schedulerapi.HostPriority{}, fmt.Errorf("node not found")
 	}
+	// get taints from node info cache.
+	taints, err := nodeInfo.Taints()
+	if err != nil {
+		// fallback to computing taints from node Spec
+		taints = node.Spec.Taints
```

Comment:
I was not suggesting that this will be visible in distributed cluster only. I was suggesting that we should check this at scale. Nodeinfo is the data structure close to scheduler(which is currently executing), so it should be fast instead of checking in the node.Spec . In fact, if you look at nodeinfo's https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/schedulercache/node_info.go#L357 - we are populating it with node.Spec.Taints which means there should be no need for us to get the taints  from nodespec again in rest of locations where we need taints. If not
- Taints can be safely removed from the node info cache, if there is no performance gain(node.Spec.taints is as good as nodeinfo.Taints(), in fact if this argument is true, there is no need to have nodeinfo cache in first place, we can calculate everything from node.Spec.xx or node.Status etc)